contract/proxy-oracle/README.md

```
@@ -49,6 +49,32 @@ Manual trip/untrip is available through governance operation `SetManualTrip { id
 Proxy oracle contract upgrades are available through the Admin-only `AdminUpgrade` operation, which dispatches the proxy's owner-gated `admin_upgrade` entrypoint. `AdminFunctionCall` is also Admin-only and dispatches a raw function call only to the governance contract's configured proxy oracle account; it is intended for exceptional proxy-admin actions such as accepting ownership after an owner transfer is proposed, not for arbitrary receiver calls. Its stored operation gas is a regular NEAR gas value; the manager CLI accepts either raw `--gas` units or `--tgas` shorthand.
 
+### Migrating a legacy NEAR proxy oracle
+
+`tmplrmgr proxy-oracle upgrade --migration v0` accepts only a v0.1.0 source. It requires the proxy-oracle account's full-access key because legacy `migrate` is private, then submits `DEPLOY_CONTRACT` and `migrate({"from_version":"v0"})` with 250 Tgas in one transaction.
+
+Before mainnet execution, refresh and run the production-state fixture:
+
+```text
+cargo test -p templar-proxy-oracle-near-contract --test migrate_mainnet generate_mainnet_state_patch -- --ignored
+cargo test -p templar-proxy-oracle-near-contract --test migrate_mainnet migrate_mainnet_patch_exactly
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)